packages/local-environment/src/network.ts

```diff
@@ -340,15 +359,164 @@ export class HardhatNetwork extends Network {
+
+  async down(): Promise<void> {
+    await this.connect();
+
+    await this.stop();
+    await this.disconnect();
+  }
+}
+export class ForkedNetwork extends Network {
+  blockTime: number;
+  keys: Key[];
+  blockExplorer: string;
+  firstStart: boolean;
+  handler: DockerizedNetworkActor;
+  weth: string;
+  forkUrl: string;
+
+  constructor(name: string, domain: number, forkUrl: string, wETHAddress: string, docker?: Dockerode, blockExplorer?: string, blocktime?: number, options?: HardhatNetworkOptions) {
+    super(name, domain, domain, docker);
+    if (blocktime) {
+      this.blockTime = blocktime;
+    } 
+    else this.blockTime = 10;
+    this.handler = new DockerizedNetworkActor(this.name, this.docker);
+    if (blockExplorer) {
+      this.blockExplorer = blockExplorer;
+    } else {
+      this.blockExplorer = "https://etherscan.io";
+    }
+    this.weth = wETHAddress;
+    this.keys = options?.keys || [];
+    this.firstStart = false;
+    this.rpc = [`http://localhost:${this.handler.port}`];
+    this.forkUrl = forkUrl;
+  }
+
+  async deployWETH(): Promise<string> {
+    return this.weth;
+  };
+
+  setWETH(wethAddress: string): string {
+    this.weth = wethAddress;
+    return this.weth;
+  }
+
+  addKeys(...ks: Key[]): void {
+    this.keys.push(...ks);
+  }
+
+  get rpcs(): string[] {
+    return this.rpc;
+  }
+
+  get specs(): NetworkSpecs {
+    return {
+      chainId: this.chainId,
+      finalizationBlocks: 2,
+      blockTime: this.blockTime,
+      supports1559: true,
+      confirmations: 2,
+      blockExplorer: this.blockExplorer,
+      indexPageSize: 2000,
+    };
+  }
+
+  get config(): ContractConfig {
+    return {
+      optimisticSeconds: 18,
+      governance: {
+        recoveryManager: this.recoveryManager,
+        recoveryTimelock: 86400,
+      },
+      updater: this.updater,
+      watchers: [this.watcher],
+    };
+  }
+
+  get bridgeConfig(): BridgeConfiguration {
+    return {
+      weth: this.weth,
+      customs: [],
+      // mintGas: 200000,
+      // deployGas: 850000
+    };
+  }
+
+  async deployToken(
+    contractFactory: ethers.ContractFactory,
+    from: string | ethers.providers.JsonRpcSigner,
+    ...args: string[]
+  ): Promise<ethers.Contract> {
+    let signer: ethers.providers.JsonRpcSigner;
+    if (typeof from === "string") {
+      signer = this.getJsonRpcSigner(from);
+    } else {
+      signer = from;
+    }
+
+    contractFactory = contractFactory.connect(signer);
+
+    const contract = await contractFactory.deploy(...args);
+
+    await contract.deployed();
+    return contract;
+  }
+
+  getJsonRpcSigner(
+    addressOrIndex: string | number
+  ): ethers.providers.JsonRpcSigner {
+    const provider = this.getJsonRpcProvider();
+    return provider.getSigner(addressOrIndex);
+  }
+
+  getJsonRpcProvider(): ethers.providers.JsonRpcProvider {
+    return new ethers.providers.JsonRpcProvider(
+      this.rpc[0]
+    );
+  }
+
+  async isConnected(): Promise<boolean> {
```

Comment:
Does this function need to be async? I noticed we're not awaiting `this.handle.isConnected()` below

Reply:
Removing this